clerk/client/rest/query.go

```diff
@@ -215,6 +215,21 @@ func recordListHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
 				return
 			}
 
+			node, err := cliCtx.GetNode()
+			if err != nil {
+				return
+			}
+
+			status, err := node.Status()
+			if err != nil {
+				return
+			}
+
+			time := status.SyncInfo.LatestBlockTime
+			if toTime > time.Unix() {
+				return
```

Comment:
Can we add a log or metric here for visibility?

Reply:
A log wouldn't be necessary. Heimdall logs update crazy fast when its syncing. A user will be able to understand the same. What say ?

Reply:
I was thinking how we can get alerted when this happens. Just realized that we have a warning message [in bor](https://github.com/maticnetwork/bor/blob/develop/consensus/bor/heimdall/client.go#L177) when it is not getting response from heimdall. I am fine with this PR then. Thanks!